### PR TITLE
NAS-109689 / 12.0 / Add warning message for Log vdev

### PR DESCRIPTION
**Pull request by denysbutenko**
image

### CODE REVIEW

**Review by william-gr — CHANGES_REQUESTED** [review]
There are two things to consider.

For data vdev the old text should remain the same with same color.

The new text and should be used for Log vdev only.

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ manager_disknumExtendConfirmMessage : T("It is not recommended to extend a pool
 
 manager_vdevtypeErrorMessage : T("Adding data vdevs of different types is not supported."),
 
-manager_stripeVdevTypeErrorMessage : T("vdev is highly discouraged and will result in data loss if it fails"),
+manager_stripeVdevTypeErrorMessage : T("vdev may result in data loss if it fails combined with a power outage."),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts:24]: small rework suggestion to put the danger first.
```suggestion
manager_stripeVdevTypeErrorMessage : T("Data loss can occur when the vdev fails and the system encounters a power outage!"),
```

**Review by Mrt134 — APPROVED** [review]
added a minor suggestion

File: src/app/helptext/storage/volumes/manager/manager.ts (hunk 1 above)
Comment (denysbutenko) [reply src/app/helptext/storage/volumes/manager/manager.ts:24]: > There are two things to consider.
> 
> For data vdev the old text should remain with the same color.
> 
> The new text and color should be used for Log vdev only.

If we add a new independent warning message that appears on `log vdev` type and can be visible with the existing message at the same time. For example, Data Vdev has 1 disk and we will see the red message then Log Vdev has 1 disk and we will see the orange message like this, does it look good enough?:
Screenshot 2021-04-01 at 14 16 56

Comment (denysbutenko) [reply src/app/helptext/storage/volumes/manager/manager.ts:24]: @Mrt134 looks good, thanks for the suggestion. one thing that I wondering it is just a part of the warning message. Full message would be `A stripe log vdev may result in data loss if it fails combined with a power outage.` and `A stripe {type} vdev is highly discouraged and will result in data loss if it fails` like on the screenshot above.

Comment (william-gr) [reply src/app/helptext/storage/volumes/manager/manager.ts:24]: LGTM

Comment (denysbutenko) [reply src/app/helptext/storage/volumes/manager/manager.ts]: @Mrt134 there are two similar messages now. Could you take a look, please?

**Comment by denysbutenko** [comment]
The requested changes were addressed. @william-gr, could you review it again?

**Review by william-gr — APPROVED** [review]
Thanks!

**Pull request merged**